NPE in LocationManager

I had this when looking over https://github.com/criticalmaps/criticalmaps-android/pull/146 - needs further investigation - placing it here to track and not forget
 
```
E/AndroidRuntime(15339): java.lang.RuntimeException: Unable to resume activity {de.stephanlindauer.criticalmaps/de.stephanlindauer.criticalmaps.Main}: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean android.location.LocationManager.isProviderEnabled(java.lang.String)' on a null object reference
E/AndroidRuntime(15339): 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4032)
E/AndroidRuntime(15339): 	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4063)
E/AndroidRuntime(15339): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3259)
E/AndroidRuntime(15339): 	at android.app.ActivityThread.access$1000(ActivityThread.java:197)
E/AndroidRuntime(15339): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1681)
E/AndroidRuntime(15339): 	at android.os.Handler.dispatchMessage(Handler.java:102)
E/AndroidRuntime(15339): 	at android.os.Looper.loop(Looper.java:145)
E/AndroidRuntime(15339): 	at android.app.ActivityThread.main(ActivityThread.java:6872)
E/AndroidRuntime(15339): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(15339): 	at java.lang.reflect.Method.invoke(Method.java:372)
E/AndroidRuntime(15339): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1404)
E/AndroidRuntime(15339): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1199)
E/AndroidRuntime(15339): Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean android.location.LocationManager.isProviderEnabled(java.lang.String)' on a null object reference
E/AndroidRuntime(15339): 	at de.stephanlindauer.criticalmaps.utils.LocationUtils.getBestLastKnownLocation(LocationUtils.java:16)
E/AndroidRuntime(15339): 	at de.stephanlindauer.criticalmaps.managers.LocationUpdateManager.getLastKnownLocation(LocationUpdateManager.java:85)
E/AndroidRuntime(15339): 	at de.stephanlindauer.criticalmaps.fragments.MapFragment.setInitialMapLocation(MapFragment.java:129)
E/AndroidRuntime(15339): 	at de.stephanlindauer.criticalmaps.fragments.MapFragment.onActivityCreated(MapFragment.java:123)
E/AndroidRuntime(15339): 	at android.support.v4.app.Fragment.performActivityCreated(Fragment.java:1970)
E/AndroidRuntime(15339): 	at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1092)
E/AndroidRuntime(15339): 	at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1248)
E/AndroidRuntime(15339): 	at android.support.v4.app.BackStackRecord.run(BackStackRecord.java:738)
E/AndroidRuntime(15339): 	at android.support.v4.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1613)
E/AndroidRuntime(15339): 	at android.support.v4.app.FragmentController.execPendingActions(FragmentController.java:330)
E/AndroidRuntime(15339): 	at android.support.v4.app.FragmentActivity.onResume(FragmentActivity.java:441)
E/AndroidRuntime(15339): 	at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1271)
E/AndroidRuntime(15339): 	at android.app.Activity.performResume(Activity.java:6671)
E/AndroidRuntime(15339): 	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4021)
E/AndroidRuntime(15339): 	... 11 more
```

--- a/app/src/main/java/de/stephanlindauer/criticalmaps/managers/LocationUpdateManager.java
+++ b/app/src/main/java/de/stephanlindauer/criticalmaps/managers/LocationUpdateManager.java
@@ -31,7 +31,6 @@ public class LocationUpdateManager {
     private static final long LOCATION_REFRESH_TIME = 12 * 1000; //12 seconds
 
     //misc
-    private final App app;
     private LocationManager locationManager;
     private SharedPreferences sharedPreferences;
     private boolean isRegisteredForLocationUpdates;
@@ -39,14 +38,13 @@ public class LocationUpdateManager {
 
     @Inject
     public LocationUpdateManager(App app, OwnLocationModel ownLocationModel, EventBusProvider eventService) {
-        this.app = app;
         this.ownLocationModel = ownLocationModel;
         this.eventService = eventService;
         sharedPreferences = app.getSharedPreferences("Main", Context.MODE_PRIVATE);
+        locationManager = (LocationManager) app.getSystemService(Context.LOCATION_SERVICE);
     }
 
     public void initializeAndStartListening() {
-        locationManager = (LocationManager) app.getSystemService(Context.LOCATION_SERVICE);
         registerLocationListeners();
     }
 

